src/main/java/de/hysky/skyblocker/skyblock/dungeon/CroesusProfit.java

```diff
@@ -0,0 +1,272 @@
+import org.apache.logging.log4j.Logger;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.*;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class CroesusProfit extends ContainerSolver {
+    private static final Logger LOGGER = LogManager.getLogger();
+    private static final Pattern ESSENCE_PATTERN = Pattern.compile("(?<type>[A-Za-z]+) Essence x(?<amount>[0-9]+)");
+    private static final Pattern CHEST_PATTERN = Pattern.compile("literal\\{(.*?) Chest}");
+    public CroesusProfit() {
+        super(".*Catacombs - Floor.*");
+    }
+
+    @Override
+    protected boolean isEnabled() {
+        return SkyblockerConfigManager.get().locations.dungeons.croesusProfit;
+    }
+
+    @Override
+    protected List<ColorHighlight> getColors(String[] groups, Map<Integer, ItemStack> slots) {
+        List<ColorHighlight> highlights = new ArrayList<>();
+        ItemStack bestChest = null, secondBestChest = null;
+        long bestValue = 0, secondBestValue = 0;    // If negative value of chest - it is out of the question
+        long dungeonKeyPriceData = getItemPrice("DUNGEON_CHEST_KEY")*2; // lesser ones don't worth the hassle
+
+        for (Map.Entry<Integer, ItemStack> entry : slots.entrySet()) {
+            ItemStack stack = entry.getValue();
+            if (stack != null && stack.getNbt() != null && stack.getName().toString().contains("Chest")) {
+                long value = valueChest(stack);
+                if (value > bestValue) {
+                    secondBestChest = bestChest;
+                    secondBestValue = bestValue;
+                    bestChest = stack;
+                    bestValue = value;
+                } else if (value > secondBestValue) {
+                    secondBestChest = stack;
+                    secondBestValue = value;
+                }
+            }
+        }
+
+        for (Map.Entry<Integer, ItemStack> entry : slots.entrySet()) {
+            ItemStack stack = entry.getValue();
+            if (stack != null && stack.getNbt() != null) {
+                if (stack.equals(bestChest)) {
+                    highlights.add(ColorHighlight.green(entry.getKey()));
+                } else if (stack.equals(secondBestChest) && secondBestValue > dungeonKeyPriceData) {
+                    highlights.add(ColorHighlight.yellow(entry.getKey()));
+                }
+            }
+        }
+        return highlights;
+    }
+
+
+    private long valueChest(@NotNull ItemStack chest) {
+        long chestValue = 0;
+        int chestPrice = 0;
+        List<String> chestItems = new ArrayList<>();
+
+        Matcher matcher = CHEST_PATTERN.matcher(chest.getName().toString());
+        String chestType = (matcher.find() ? matcher.group(1) : "Error");     // we are doing it for sole reason of saving us headache of a spirit pet rarity
+
+        boolean processingContents = false;
+        for (Text line : chest.getTooltip(null, TooltipContext.BASIC)) {
```

Comment:
Use `ItemUtils#getNbtTooltips` instead as this method returns additional. unneeded information.